projects/storefrontlib/src/cms-components/myaccount/my-coupons/my-coupons.component.service.ts

```diff
@@ -0,0 +1,27 @@
+import { Injectable } from '@angular/core';
+import { CustomerCoupon, RoutingService } from '@spartacus/core';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class MyCouponsComponentService {
```

Comment:
since we use component service, we could move more logic here, so the component can be ligther

Reply:
move some logic(getSortLabels) to this service to make the component lighter. The others called from CustomerCouponService remains due to add a wrapper doesn't have much meaning.